middlewares/osio/auth.go

```diff
@@ -257,19 +261,36 @@ func (r RequestType) path() string {
 
 func (r RequestType) getTargetURL(ns namespace) string {
 	switch r {
-	case api:
+	case api, oapi:
 		return ns.ClusterURL
 	case metrics:
 		return ns.ClusterMetricsURL
 	case console:
 		return ns.ClusterConsoleURL
 	case logs:
 		return ns.ClusterLoggingURL
+	case undefine:
+		return ns.ClusterURL
 	default:
 		return ns.ClusterURL
 	}
 }
 
+func (r RequestType) stripPathPrefix(req *http.Request) {
+	var pathPrefix, stripPath string
+	switch r {
+	case api, oapi:
+		pathPrefix = r.path() + r.path()
+		stripPath = r.path()
+	case metrics, console, logs:
+		pathPrefix = r.path()
+		stripPath = r.path()
+	case undefine:
+		return
+	}
+	stripRequestPathPrefix(req, pathPrefix, stripPath)
+}
```

Comment:
strip path logic made more flexible.  depending on RequestType pathPrefix and stripPath value determine.  Now stripRequestPathPrefix() takes two args.